src/Client/Adapter/Curl.php

```diff
@@ -195,17 +195,33 @@ public function connect($host, $port = 80, $secure = false)
             curl_setopt($this->curl, CURLOPT_PORT, intval($port));
         }
 
-        if (isset($this->config['timeout'])) {
+        if (isset($this->config['connecttimeout'])) {
+            $connectTimeout = $this->config['connecttimeout'];
+        } elseif (isset($this->config['timeout'])) {
+            $connectTimeout = $this->config['timeout'];
+        } else {
+            $connectTimeout = null;
+        }
+        if ($connectTimeout !== null) {
             if (defined('CURLOPT_CONNECTTIMEOUT_MS')) {
-                curl_setopt($this->curl, CURLOPT_CONNECTTIMEOUT_MS, $this->config['timeout'] * 1000);
+                curl_setopt($this->curl, CURLOPT_CONNECTTIMEOUT_MS, $connectTimeout * 1000);
             } else {
-                curl_setopt($this->curl, CURLOPT_CONNECTTIMEOUT, $this->config['timeout']);
+                curl_setopt($this->curl, CURLOPT_CONNECTTIMEOUT, $connectTimeout);
             }
+        }
 
+        if (isset($this->config['executetimeout'])) {
```

Comment:
Instead of add a new parameter `executetimeout` why we don't use the old one `timeout`? The meaning is the same and we can be aligned with the curl constant naming.